### ISSUE TITLE
Detect dependencies used for types

### ISSUE THREAD

**Issue by koterpillar** [tweag/FawltyDeps#221]
**Is your feature request related to a problem? Please describe.**

My project uses:
* `requests` (as an example)
*  `mypy` for type checking
* and therefore [`types-requests`](https://pypi.org/project/types-requests/) from [typeshed](https://github.com/python/typeshed/).

FawltyDeps reports `types-requests` as unused, together with other typeshed packages.

**Describe the solution you'd like**

I'd like FawltyDeps to consider `types-ABCD` used if and only if `ABCD` is used (or re-use the logic from `mypy` in case there are non-trivial rules for which packages declare stubs for which).

This can be behind a flag (e.g. `typeshed = true`) in configuration.

**Describe alternatives you've considered**

I can manually ignore the `types-*` packages using `ignore_unused`, but then if `requests` is removed `types-requests` will need to be removed manually, and if one forgets that it'll become fawlty (tm?) - the problem this project is trying to prevent :-)

**Comment by jherland** [comment]
Agreed. We do not yet handle these cases as well as we should. There is a related issue (https://github.com/tweag/FawltyDeps/issues/117) that focuses on tools (like `black`, `pylint`, etc.) that are never intended to be imported, but these supplementary modules for improved type checking have escaped our attention until now.

I'll update the last item on our roadmap (https://github.com/tweag/FawltyDeps/issues/195) to include this issue as well. I can't promise exactly when we'll get to it, but now we know about it.

Thanks for submitting!

**Comment by Nour-Mws** [comment]
I recall us having a discussion about `types-setuptools`, which we declare in FawltyDeps and which is possibly related to this issue, but I couldn't find that discussion on or off Github (see also https://github.com/tweag/FawltyDeps/issues/117#issuecomment-1424225038).
